errors/errors.go

```diff
@@ -0,0 +1,22 @@
+package errors
+
+import (
+	"encoding/hex"
+	"fmt"
+)
+
+// NoSuchObject is an error type that occurs when no object with a given object
+// ID is available.
+type NoSuchObject struct {
```

Comment:
Another thing we might think about is making this name `NoSuchObject` the constructor for an otherwise-private implementation, which I think would read nicely when returning an error.

For example, instead of:

```go
return nil, &errors.NoSuchObject{Oid: oid}
```

we could write:

```go
return nil, errors.NoSuchObject(oid)
```